### PR TITLE
Feature/mentor mentee update endpoints

### PR DESCRIPTION
**Pull request by GermanParra**
## Description

https://www.loom.com/share/d388d6abf5954691a7b2fd133055a7e8

Create /update/mentors and /update/mentees endpoints, implementing Pydantic schemas to validate the changes.

## Type of change

- [x] New feature

## Checklist:

- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by BrandonSmith710 — APPROVED** [review]
Nice job German. Really good documentation too.

**Review by AdamHVu — APPROVED** [review]
Schema is clean and update endpoints work!

**Review by yuririnnnu — APPROVED** [review]

**Comment by VABIII** [comment]
Code looks sharp, and performs as expected

**Review by VABIII — APPROVED** [review]
Code looks sharp, and performs as expected.

File: app/api.py [hunk 1]
```diff
@@ -63,6 +63,42 @@ async def computer_assignment_rating_visualizer():
     return computer_assignment_visualizer(API.db)
 
 
+@API.post("/update/mentors")
+async def update(query: Dict, update_data: Dict):
```
Comment (jinjahninjah) [inline app/api.py:67]: This is going to throw an error as the route and function have the same name. Instead, rename the function to update_mentors or something unique.

File: app/api.py [hunk 2]
```diff
@@ -63,6 +63,42 @@ async def computer_assignment_rating_visualizer():
...
+        update_data (dict): Key value pairs to update
+    Returns:
+        Updated fields or schema discrepancy error as dictionary
+    """
+    MentorUpdate(**update_data)
+    return {"result": API.db.update("Mentors", query, update_data)}
+
+
+@API.post("/update/mentees")
+async def update(query: Dict, update_data: Dict):
```
Comment (jinjahninjah) [inline app/api.py:85]: same thing here, change the function name to something else and you should be good to go 👍

**Review by jinjahninjah — CHANGES_REQUESTED** [review]
Great work in refactoring the endpoints to use Pydantic! The only change would be the flagged functions and it should be good to go 👍

File: app/api.py (hunk 1 above)
Comment (mikayla-kosmala) [reply app/api.py:67]: Would that be the same for the other pull tequests about endpoints?

Comment (GermanParra) [reply app/api.py]: Thank you, changes ready! 👍

**Comment by jinjahninjah** [comment]
Thanks for importing the MentorUpdate and MenteeUpdate classes as well! 👍

**Review by jinjahninjah — APPROVED** [review]

**Review by cbell98 — APPROVED** [review]
Pydantic check. Schema check. Endpoints check.

**Pull request merged**